resource_customizations/minio.min.io/Tenant/health.lua

```diff
@@ -0,0 +1,46 @@
+health_status = {}
+if obj.status ~= nil then
+    if obj.status.currentState ~= nil then
+        if obj.status.currentState == "Initialized" then
+            health_status.status = "Healthy"
+            health_status.message = obj.status.currentState
+            return health_status
+        end
+        if obj.status.currentState:find("^Provisioning") ~= nil then
+            health_status.status = "Progressing"
+            health_status.message = obj.status.currentState
+            return health_status
+        end
+        if obj.status.currentState:find("^Waiting") ~= nil then
+            health_status.status = "Progressing"
+            health_status.message = obj.status.currentState
+            return health_status
+        end
+        if obj.status.currentState:find("^Updating") ~= nil then
+            health_status.status = "Progressing"
+            health_status.message = obj.status.currentState
+            return health_status
+        end
+        if obj.status.currentState == "Statefulset not controlled by operator" then
+            health_status.status = "Degraded"
+            health_status.message = obj.status.currentState
+            return health_status
+        end
+        if obj.status.currentState == "Another MinIO Tenant already exists in the namespace" then
+            health_status.status = "Degraded"
+            health_status.message = obj.status.currentState
+            return health_status
+        end
+        if obj.status.currentState == "Different versions across MinIO Pools" then
+            health_status.status = "Degraded"
+            health_status.message = obj.status.currentState
+            return health_status
+        end
+        health_status.status = "Unknown"
+        health_status.message = obj.status.currentState
+        return health_status
+    end
+end
+health_status.status = "Missing"
```

Comment:
Health check should not return `Unknown` or `Missing`. These are internal statuses. `Unknown` means error and `Missing` means expected resource not found in the cluster. To be consistent with other checks the health check need to return `Progressing`

Reply:
Fixed!
Could status descriptions be added to the [documentation](https://argo-cd.readthedocs.io/en/stable/operator-manual/health/)?

Reply:
Agree. Created PR with docs changes: https://github.com/argoproj/argo-cd/pull/6692